my-first-radanalytics-app.adoc

```diff
@@ -46,25 +60,57 @@ pass:[<img src="/assets/my-first-radanalytics-app/sparkpi-architecture.svg" clas
-
-As https://docs.openshift.org/latest/architecture/core_concepts/builds_and_image_streams.html#source-build[source-to-image] technology is fundamental to the operation of this tutorial, let's take a closer look at its details. Source-to-image provides a convenient way for you to produce container images directly from your source repositories. This work is handled by language specific builder images which can ingest your source code, assemble it and then produce an application image which is ready to be deployed. In the case of the Oshinko source-to-image builders, an Apache Spark cluster will also be created dynamically for your application when it runs.
+The SparkPi component is the main component that you will be concerned with
+throughout the tutorial. You should already have a computer with the necessary
+tools installed for making Pi requests, editing your source files and
+commanding OpenShift to deploy your microservice. The Apache Spark cluster is
+created automatically for you by the
+http://github.com/radanalyticsio/oshinko-s2i[Oshinko source-to-image tools].
+
+https://docs.openshift.org/latest/architecture/core_concepts/builds_and_image_streams.html#source-build[Source-to-image]
+technology is fundamental to the operation of this tutorial. Source-to-image
+provides a convenient way for you to produce container images directly from
+your source repositories. This work is handled by language specific builder
+images which can ingest your source code, assemble it and then produce an
+application image which is ready to be deployed. In the case of the Oshinko
+source-to-image builders, an Apache Spark cluster is also created dynamically
+for your application when it runs.
 
 [[setup]]
 == Setup your coding environment
 
-To utilize the source-to-image workflow you will need to have a https://git-scm.com[Git] repository to store your work. This repository is then used by the source-to-image builders to create the final application image which will be run on OpenShift. For the purposes of this tutorial your repository will need to be accessible by the OpenShift cluster. The easiest way to do this is by using an external service like https://github.com[GitHub] or https://bitbucket.com[BitBucket].
+To use the source-to-image workflow you will need to have a
+https://git-scm.com[Git] repository to store your work. This repository is
+then used by the source-to-image builders to create the final application
+image which will be run on OpenShift. For the purposes of this tutorial your
+repository must be accessible by the OpenShift cluster. The easiest way to do
+this is by using an external service like https://github.com[GitHub],
+https://bitbucket.com[BitBucket], or https://gitlab.com[GitLab].
```

Comment:
Bitbucket was removed previously, should it be added?

Reply:
not sure i follow on this comment, bitbucket never got removed from this tutorial, am i missing something?

Reply:
My fault, from the diff it looked that Bitbucket had been removed from https://github.com/radanalyticsio/radanalyticsio.github.io/pull/260/commits/06dac67f0a5d5403568f10ae1f468dec90aebe19#diff-91bfd9db64a5087f636915c30630efa9L26 !